cookbooks/bcpc-hadoop/recipes/resource_manager.rb

```diff
@@ -32,22 +32,54 @@
   not_if "hdfs dfs -test -d /var/log/hadoop-yarn", :user => "hdfs"
 end
 
+bash "create-hdfs-yarn-ats-log" do
+  code <<-EOH
+  hdfs dfs -mkdir -p /var/log/ats && hdfs dfs -chmod 1777 /var/log/ats && hdfs dfs -chown yarn:mapred /var/log/ats
+  hdfs dfs -mkdir -p /var/log/ats/active && hdfs dfs -chmod 1777 /var/log/ats/active && hdfs dfs -chown yarn:mapred /var/log/ats/active
+  hdfs dfs -mkdir -p /var/log/ats/done && hdfs dfs -chmod 0700 /var/log/ats/done && hdfs dfs -chown yarn:mapred /var/log/ats/done
+  EOH
+  user "hdfs"
+  only_if { node.roles.include?("BCPC-Hadoop-Head-YarnTimeLineServer") }
+  not_if "hdfs dfs -test -d /var/log/hadoop-yarn/ats/active && hdfs dfs -test -d /var/log/hadoop-yarn/ats/done", :user => "hdfs"
+end
+
 # list hdp packages to install
-%w{hadoop-yarn-resourcemanager hadoop-client hadoop-mapreduce-historyserver}.each do |pkg|
+yarn_packages = %w{hadoop-yarn-resourcemanager hadoop-client hadoop-mapreduce-historyserver}
+if node.roles.include?("BCPC-Hadoop-Head-YarnTimeLineServer")
+  yarn_packages.push("hadoop-yarn-timelineserver")
+end
+
+yarn_packages.each do |pkg|
   package hwx_pkg_str(pkg, node[:bcpc][:hadoop][:distribution][:release]) do
     action :install
   end
-
   hdp_select(pkg, node[:bcpc][:hadoop][:distribution][:active_release])
 end
 
+package hwx_pkg_str("tez", node[:bcpc][:hadoop][:distribution][:release]) do
+  action :install
+  only_if { node.roles.include?("BCPC-Hadoop-Head-YarnTimeLineServer") }
```

Comment:
Instead of all the `only_if`'s since recipes must be known before we start converging you could wrap all this in a ruby if block too right?